FlashDevelop/Managers/ShortcutManager.cs

```diff
@@ -4,233 +4,319 @@
+                var keys = Globals.MainForm.GetShortcutItemKeys(id);
+                if (!keys.IsNone)
                 {
+                    if (item is ToolStripMenuItemEx)
+                    {
+                        var casted = item as ToolStripMenuItemEx;
+                        if (casted.ShortcutKeys.IsNone)
+                        {
+                            casted.ShortcutKeyDisplayString = view ? keys.ToString() : null;
+                        }
+                    }
                     if (item is ToolStripMenuItem)
                     {
                         var casted = item as ToolStripMenuItem;
                         if (casted.ShortcutKeys == Keys.None)
                         {
-                            String keytext = DataConverter.KeysToString(keys);
-                            casted.ShortcutKeyDisplayString = view ? keytext : "";
+                            casted.ShortcutKeyDisplayString = view ? keys.ToString() : null;
                         }
                     }
                     else
                     {
-                        Int32 end = item.ToolTipText.IndexOfOrdinal(" (");
-                        String keytext = view ? " (" + DataConverter.KeysToString(keys) + ")" : "";
+                        int end = item.ToolTipText.LastIndexOfOrdinal(" (");
+                        string keytext = view ? " (" + keys + ")" : string.Empty;
                         if (end != -1) item.ToolTipText = item.ToolTipText.Substring(0, end) + keytext;
-                        else item.ToolTipText = item.ToolTipText + keytext;
+                        else item.ToolTipText += keytext;
                     }
                 }
             }
         }
 
         /// <summary>
-        /// Loads the custom shortcuts from a file
+        /// Loads the custom shortcuts from a file.
         /// </summary>
         public static void LoadCustomShortcuts()
         {
             ScintillaControl.InitShortcuts();
-            String file = FileNameHelper.ShortcutData;
-            if (File.Exists(file))
+            string file = FileNameHelper.ShortcutData;
+            if (!File.Exists(file))
             {
-                List<Argument> shortcuts = new List<Argument>();
-                shortcuts = (List<Argument>) ObjectSerializer.Deserialize(file, shortcuts, false);
-                foreach (Argument arg in shortcuts)
+                if (File.Exists(FileNameHelper.ShortcutDataOld))
                 {
-                    ShortcutItem item = GetRegisteredItem(arg.Key);
-                    if (item != null) item.Custom = (Keys) Enum.Parse(typeof(Keys), arg.Value);
+                    File.Move(FileNameHelper.ShortcutDataOld, file);
+                }
+                else return;
+            }
+            var shortcuts = new List<Argument>();
+            shortcuts = (List<Argument>) ObjectSerializer.Deserialize(file, shortcuts, false);
```

Comment:
The method is called inside a `try/catch` block already, inside `MainForm.InitializeMainForm()`.